uaclient/tests/test_gpg.py

```diff
@@ -0,0 +1,70 @@
+from uaclient import exceptions, gpg, util
+from uaclient.testing import data
+
+import logging
+import os
+import pytest
+
+
+class TestExportGPGKeyFromKeyring:
+
+    def test_key_error_on_missing_key(self, tmpdir):
+        """Raise UserFacingError when key_id is not in source_keyring_file."""
+        src_keyring = tmpdir.join('ubuntu-advantage-keyring')
+        src_keyring.write('')
+        destination_keyfile = tmpdir.join('destination_keyfile').strpath
+        key_id = 'Bueller... Bueller'
```

Comment:
Should we make this a valid key ID, so we're sure we aren't hitting a GPG code path that tests for ID validity?